ray-operator/controllers/ray/common/pod.go

```diff
@@ -264,11 +263,17 @@ func DefaultWorkerPodTemplate(instance rayv1alpha1.RayCluster, workerSpec rayv1a
 	return podTemplate
 }
 
-func initLivenessProbeHandler(probe *v1.Probe, rayNodeType rayv1alpha1.RayNodeType) {
-	if probe.Exec == nil {
-		// we only create the probe if user did not specify any.
+// For KubeRay, the liveness and readiness probes perform the same checks.
+// Hence, we use the same function to initialize both probes.
+func initHealthProbe(probe *v1.Probe, rayNodeType rayv1alpha1.RayNodeType) {
+	if probe.Exec == nil && probe.HTTPGet == nil && probe.TCPSocket == nil && probe.GRPC == nil {
+		// Case 1: head node => Check GCS and Raylet status.
+		// Case 2: worker node => Check Raylet status.
+		//
+		// Note: Since the Raylet process and the dashboard agent process are fate-sharing,
+		// we only need to check one of them.
```

Comment:
It would be helpful to write here which one we're checking (raylet process or dashboard agent process).  It's not immediately obvious from the code (because it has both `DefaultDashboardAgentListenPort, RayAgentRayletHealthPath`)

Reply:
The probe uses the dashboard agent's API to check the status of Raylet.

Reply:
Update: https://github.com/ray-project/kuberay/pull/1364/commits/2e2661e405fccf525c1f422c7056df3f687bc0eb